Extension/lib/content-script/assistant/js/adguard-selector.js

```diff
@@ -0,0 +1,479 @@
+
+        return false;
+    };
+
+    var sgMousedownHandler = function (e) {
+        e.preventDefault();
+
+        if (unbound) {
+            return true;
+        }
+
+        var elem = this;
+        var w_elem = $(elem);
+        if (w_elem.hasClass(BORDER_CLASS)) {
+            //Clicked on one of our floating borders, target the element that we are bordering.
+            elem = elem.target_elem || elem;
+        }
+
+        if (elem == document.body || elem == document.body.parentNode) {
+            return;
+        }
+
+        // Don't allow selection of elements that have a selected child.
+        if ($('.sg_selected', this).get(0)) {
+            blockClicksOn(elem);
+        }
+
+        makePredictionPath(elem);
+
+        removeBorders();
+        blockClicksOn(elem);
+
+        // Refresh the borders by triggering a new mouseover event.
+        w_elem.trigger("mouseover", {'self': this});
+
+        onElementSelectedHandler(elem);
+
+        return false;
+    };
+
+
+    var setupEventHandlers = function () {
+        makeIFrameAndEmbededSelector();
+
+        var sgIgnore = $("body *:not(.sg_ignore)");
+        sgIgnore.on("mouseover", {'self': this}, sgMouseoverHandler);
+        sgIgnore.on("mouseout", {'self': this}, sgMouseoutHandler);
+        sgIgnore.on("click", {'self': this}, sgMousedownHandler);
+    };
+
+    var deleteEventHandlers = function () {
+        removePlaceholders();
+
+        var elements = $("body *");
+        elements.off("mouseover", sgMouseoverHandler);
+        elements.off("mouseout", sgMouseoutHandler);
+        elements.off("click", sgMousedownHandler);
+    };
+
+    var setupBorders = function () {
+        if (!borderTop) {
+            var width = px(BORDER_WIDTH);
+
+            borderTop = $('<div>').addClass(BORDER_CLASS).css('height', width).hide()
+                .on("click", {'self': this}, sgMousedownHandler);
+            borderBottom = $('<div>').addClass(BORDER_CLASS).addClass('sg_bottom_border')
+                .css('height', px(BORDER_WIDTH + 6)).hide()
+                .bind("click", {'self': this}, sgMousedownHandler);
+            borderLeft = $('<div>').addClass(BORDER_CLASS).css('width', width).hide()
+                .on("click", {'self': this}, sgMousedownHandler);
+            borderRight = $('<div>').addClass(BORDER_CLASS).css('width', width).hide()
+                .on("click", {'self': this}, sgMousedownHandler);
+
+            addBorderToDom();
+        }
+    };
+
+
+    // PUBLIC API
+
+    /**
+     * Starts selector module.
+     *
+     * @param onElementSelected callback function
+     * @param selectionRenderer optional function contains selection presentation implementation
+     */
+    api.init = function (onElementSelected, selectionRenderer) {
+
+        onElementSelectedHandler = onElementSelected;
+        if (selectionRenderer && typeof selectionRenderer === "function") {
+            selectionRenderer = selectionRenderer;
+        }
```

Comment:
If parameter selectionRenderer is present, function selectionRenderer would not be overwritten.